js/algorithms/simple-arbitrage.js

```diff
@@ -1,63 +1,78 @@
-function outputExchangeData(allExchangeData) {
+function outputExchangeData(pair, time) {
 
   // Temporary variables for algorithm
   var highest_bid = 0;
   var highest_bid_amt = 0;
   var lowest_ask = 10000000000;
   var lowest_ask_amt = 0;
   var low_amt;
+  var data = {};
+
+  // Passing in subset constructor
+  var subSet = require("./../classes/sub-set.js");
+
+  // Creating a subset of data and opening feeds to that data
+  var Exchanges = subSet("all", [pair]);
+  Exchanges.open();
 
   // Code for parsing the maps in each exchange
   var mapParse = require("./../data-parsing/map-parsing.js");
 
-  // Condensing maps into the highest bid and lowest ask
-  var Krak_Info = mapParse(allExchangeData.kraken, "kraken");
-  var Polo_Info = mapParse(allExchangeData.poloniex, "poloniex");
-  var GDAX_Info = mapParse(allExchangeData.gdax, "gdax");
-  var Bitf_Info = mapParse(allExchangeData.bitfinex, "bitfinex");
-
-  // Creating a overall array to store highest bids and lowest asks
-  var overall = [Polo_Info, GDAX_Info, Bitf_Info, Krak_Info];
+  // Calling the actual arbitrage logic in the time interval specified
+  setInterval(function () {
+    for (var key in Exchanges) {
+      if (Exchanges.hasOwnProperty(key)) {
+        for (var key_2 in Exchanges[key]){
+          if (Exchanges[key].hasOwnProperty(key_2)) {
+            data[key] = mapParse(Exchanges[key][key_2], key);
+          }
+        }
```

Comment:
### 1.
This for loop enters into each exchange and places the mapParse return value (highest bid, amount, lowest bid, amount) into the data variable with a key corresponding to the exchange name. This function was mainly designed for single pair-multi exchange arbitrage, but overall its pretty hacky so I'll replace it.
### 2.
Good idea. I'll start on that.
### 3.
So the code I wrote takes in a array of exchanges and a array of pairs, it then constructs a data-set of those exchanges, with those pairs. So if you called 
```javascript 
var necData = SubSet(['poloniex', 'kraken'], ['BTCUSD','ETHBTC']);
```
It would return 
```
{ poloniex: 
   { BTCUSD: 
      { pair: 'USDT_BTC',
        last30Day: [],
        currTopOrderBook: [],
        OrderBook: {},
        highbids: Map {},
        lowasks: Map {} },
     ETHBTC: 
      { pair: 'BTC_ETH',
        last30Day: [],
        currTopOrderBook: [],
        OrderBook: {},
        highbids: Map {},
        lowasks: Map {} } },
  kraken: 
   { BTCUSD: 
      { pair: 'XXBTZUSD',
        last30Day: [],
        currTopOrderBook: [],
        OrderBook: {},
        highbids: Map {},
        lowasks: Map {} },
     ETHBTC: 
      { pair: 'XETHXXBT',
        last30Day: [],
        currTopOrderBook: [],
        OrderBook: {},
        highbids: Map {},
        lowasks: Map {} } },
  open: [Function] }
```